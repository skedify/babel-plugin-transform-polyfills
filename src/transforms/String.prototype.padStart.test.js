import compile from '../../test/utils/compile'

describe('String.prototype.padStart', () => {
  it('should replace String.prototype.padStart with a polyfill', () => {
    const result = compile(
      `
      const text = 'this is a string'
      if (text.padStart('a') === true) {
        console.log('success')
      } else {
        console.log('failure')
      }
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should not import polyfill more than once', () => {
    const result = compile(
      `
      const text = 'this is a string'
      if (text.padStart('a')) {
        console.log(text.padStart('a'))
      } else {
        console.log('failure')
      }
    `
    )

    expect(result).toMatchSnapshot()
  })

  it('should work when part of a call chain', () => {
    const result = compile(
      `
      const text = 'this is a string'
      if (text.substr(5).padStart('a')) {
        console.log('success')
      } else {
        console.log('failure')
      }
    `
    )

    expect(result).toMatchSnapshot()
  })
})
