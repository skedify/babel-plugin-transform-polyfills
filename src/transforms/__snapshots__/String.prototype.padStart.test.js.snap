// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String.prototype.padStart should not import polyfill more than once 1`] = `
"import _String_prototype_padStart from \\"babel-plugin-transform-polyfills/polyfills/String/prototype/padStart\\";
const text = 'this is a string';

if (_String_prototype_padStart(text, 'a')) {
  console.log(_String_prototype_padStart(text, 'a'));
} else {
  console.log('failure');
}"
`;

exports[`String.prototype.padStart should replace String.prototype.padStart with a polyfill 1`] = `
"import _String_prototype_padStart from \\"babel-plugin-transform-polyfills/polyfills/String/prototype/padStart\\";
const text = 'this is a string';

if (_String_prototype_padStart(text, 'a') === true) {
  console.log('success');
} else {
  console.log('failure');
}"
`;

exports[`String.prototype.padStart should work when part of a call chain 1`] = `
"import _String_prototype_padStart from \\"babel-plugin-transform-polyfills/polyfills/String/prototype/padStart\\";
const text = 'this is a string';

if (_String_prototype_padStart(text.substr(5), 'a')) {
  console.log('success');
} else {
  console.log('failure');
}"
`;
